packages/k4marlinwrapper/package.py

```diff
@@ -33,6 +33,7 @@ class K4marlinwrapper(CMakePackage, Ilcsoftpackage):
     depends_on('edm4hep')
     depends_on('k4lcioreader')
     depends_on('wget', type=('test'))
+    depends_on('catch2@3.0.1:', when='@0.3.2:', type=('build', 'test'))
```

Comment:
```suggestion
    depends_on('catch2@3.0.1:', when='@0.3.2:', type=('test'))
```
I just noticed this is also wrong in other recipes. The `test` dependency is essentially a `build` dependency, conditional on the `--test` argument  being present. So, `type=('build', 'test')` makes no sense, since it is completely identical to `type=('build')`.

Slightly related, for all the framework packages the definitions  of dependency types in spack are not really ideal - it would be much easier to run the test if 'test' was both 'build' and 'run' - but that requires some more thought, for now we manage by manually setting the run environment in ctest.

Reply:
Ah, I see. I have just copy/pasted from k4SimDelphes ;)

I am not sure I agree with `test` ideally being `build` and `run`. In my opinion a package should be able to be built and tested without any additional environments setup (compared to `build`) and without having to install the package first. So I think, managing the environment explicitly in ctest is the way to go, at least for the packages that we maintain ourselves. Or am I confusing things here with the different spack phases?